### ISSUE TITLE
Error on calling any API: [SDK.CanNotResolveEndpoint] Can not resolve endpoint

### ISSUE THREAD

**Issue by jancco** [aliyun/alibaba-cloud-sdk-go#136]
After pulling the latest version, our all codes when calling API get the following error:

> Error: [SDK.CanNotResolveEndpoint] Can not resolve endpoint(param = {"Domain":"","Product":"Ecs","RegionId":"cn-shanghai","LocationProduct":"ecs","LocationEndpointType":"openAPI"}), please check your accessKey with secret, and read the user guide

In order to reappear this issue, I wrote a test case as following:

```go
import (
	"fmt"
	"testing"
	aliecs "github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
)

var RegionId, AccessKeyId, SecretAccessKey string

func init() {
	RegionId = "cn-beijing"
	AccessKeyId = "***"
	SecretAccessKey = "***"
}

func TestDescribeRegions(t *testing.T) {
	input := aliecs.CreateDescribeRegionsRequest()
	output, err := getEcsClient().DescribeRegions(input)
	if err != nil {
		fmt.Printf("Failed to DescribeRegions.! Error: %s\n", err.Error())
		return
	}

	fmt.Printf("len(Regions): %d\n", len(output.Regions.Region))
}

func getEcsClient() *aliecs.Client {
	client, err := aliecs.NewClientWithAccessKey(RegionId, AccessKeyId, SecretAccessKey)
	printOperationError("NewClientWithAccessKey", err)
	return client
}

func printOperationError(operation string, err error) {
	if err != nil {
		fmt.Printf("Failed to %s! Error: %s\n", operation, err.Error())
	}
}
```

Please help checking what's the wrong. Thanks!

**Comment by JacksonTian** [comment]
fixed in 2fb543b0d6368336297973544b0b653a1f306307

**Comment by JacksonTian** [comment]
@jehos Now we have enough test cases to make sure no breaking change.
